### ISSUE TITLE
Segfault with Text#new when font file is missing from given path

### ISSUE THREAD

**Issue by jjgh** [ruby2d/ruby2d#41]
O.S: Windows 7 

> ruby -v
ruby 2.3.3p222 (2016-11-21 revision 56859) [x64-mingw32]

```
require 'ruby2d'
set title: "Hello World!"
set width: 768, height: 300
t =  Text.new(15, 15, "test", 40, "fonts/arial.ttf")

# also happens with just 
# t = Text.new

show
```

Output:

```

C:\Users\xxx\Downloads>ruby ruby2d-hello.rb
C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d/application.rb:44: [BUG] Segmentation fault
ruby 2.3.3p222 (2016-11-21 revision 56859) [x64-mingw32]

-- Control frame information -----------------------------------------------
c:0005 p:---- s:0014 e:000013 CFUNC  :show
c:0004 p:0009 s:0011 e:000010 METHOD C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d/application.rb:44
c:0003 p:0014 s:0008 e:000007 METHOD C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d/dsl.rb:33
c:0002 p:0080 s:0005 E:0011e8 EVAL   ruby2d-hello.rb:13 [FINISH]
c:0001 p:0000 s:0002 E:000800 (none) [FINISH]

-- Ruby level backtrace information ----------------------------------------
ruby2d-hello.rb:13:in `<main>'
C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d/dsl.rb:33:in `show'
C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d/application.rb:44:in `show'
C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d/application.rb:44:in `show'

-- C level backtrace information -------------------------------------------
C:\Windows\SYSTEM32\ntdll.dll(ZwWaitForSingleObject+0xa) [0x000000007721BD7A]
C:\Windows\system32\KERNELBASE.dll(WaitForSingleObjectEx+0x9c) [0x000007FEFD0310AC]
C:\Dev\ruby\Ruby23-x64\bin\x64-msvcrt-ruby230.dll(rb_vm_bugreport+0xa4) [0x000000006E6AF354]
C:\Dev\ruby\Ruby23-x64\bin\x64-msvcrt-ruby230.dll(rb_bug_context+0x68) [0x000000006E54B6B8]
C:\Dev\ruby\Ruby23-x64\bin\x64-msvcrt-ruby230.dll(rb_check_safe_obj+0x3b2) [0x000000006E61E522]
 [0x0000000000401A85]
C:\Windows\SYSTEM32\ntdll.dll(_C_specific_handler+0x9c) [0x00000000771E7958]
C:\Windows\SYSTEM32\ntdll.dll(RtlDecodePointer+0xad) [0x00000000771F812D]
C:\Windows\SYSTEM32\ntdll.dll(RtlUnwindEx+0xbbf) [0x00000000771E855F]
C:\Windows\SYSTEM32\ntdll.dll(KiUserExceptionDispatcher+0x2e) [0x000000007721BCB8]
 [0x0000000067FC23FB]
C:\Dev\ruby\Ruby23-x64\lib\ruby\gems\2.3.0\gems\ruby2d-0.3.0\lib\ruby2d\ruby2d.so(Init_ruby2d+0x2139) [0x0000000067FC53F9]
 [0x0000000067FC3124]
C:\Dev\ruby\Ruby23-x64\bin\x64-msvcrt-ruby230.dll(rb_error_arity+0x1fb) [0x000000006E697D1B]
C:\Dev\ruby\Ruby23-x64\bin\x64-msvcrt-ruby230.dll(rb_vm_invoke_proc+0x760) [0x000000006E6A4280]
C:\Dev\ruby\Ruby23-x64\bin\x64-msvcrt-ruby230.dll(rb_vm_localjump_error+0xe11) [0x000000006E69E9B1]
C:\Dev\ruby\Ruby23-x64\bin\x64-msvcrt-ruby230.dll(rb_vm_localjump_error+0x5788) [0x000000006E6A3328]
C:\Dev\ruby\Ruby23-x64\bin\x64-msvcrt-ruby230.dll(rb_check_copyable+0x2dc1) [0x000000006E5509B1]
C:\Dev\ruby\Ruby23-x64\bin\x64-msvcrt-ruby230.dll(ruby_run_node+0x51) [0x000000006E553951]
 [0x0000000000402D14]
 [0x00000000004013D7]
 [0x00000000004014F8]
C:\Windows\system32\kernel32.dll(BaseThreadInitThunk+0xd) [0x00000000770C59CD]

-- Other runtime information -----------------------------------------------

* Loaded script: ruby2d-hello.rb

* Loaded features:

    0 enumerator.so
    1 thread.rb
    2 rational.so
    3 complex.so
    4 C:/Dev/ruby/Ruby23-x64/lib/ruby/2.3.0/x64-mingw32/enc/encdb.so
    5 C:/Dev/ruby/Ruby23-x64/lib/ruby/2.3.0/x64-mingw32/enc/trans/transdb.so
    6 C:/Dev/ruby/Ruby23-x64/lib/ruby/2.3.0/x64-mingw32/enc/windows_1252.so
    7 C:/Dev/ruby/Ruby23-x64/lib/ruby/2.3.0/unicode_normalize.rb
    8 C:/Dev/ruby/Ruby23-x64/lib/ruby/2.3.0/x64-mingw32/rbconfig.rb
    9 C:/Dev/ruby/Ruby23-x64/lib/ruby/site_ruby/2.3.0/rubygems/compatibility.rb
   10 C:/Dev/ruby/Ruby23-x64/lib/ruby/site_ruby/2.3.0/rubygems/defaults.rb
   11 C:/Dev/ruby/Ruby23-x64/lib/ruby/site_ruby/2.3.0/rubygems/deprecate.rb
   12 C:/Dev/ruby/Ruby23-x64/lib/ruby/site_ruby/2.3.0/rubygems/errors.rb
   13 C:/Dev/ruby/Ruby23-x64/lib/ruby/site_ruby/2.3.0/rubygems/version.rb
   14 C:/Dev/ruby/Ruby23-x64/lib/ruby/site_ruby/2.3.0/rubygems/requirement.rb
   15 C:/Dev/ruby/Ruby23-x64/lib/ruby/site_ruby/2.3.0/rubygems/platform.rb
   16 C:/Dev/ruby/Ruby23-x64/lib/ruby/site_ruby/2.3.0/rubygems/basic_specification.rb
   17 C:/Dev/ruby/Ruby23-x64/lib/ruby/site_ruby/2.3.0/rubygems/stub_specification.rb
   18 C:/Dev/ruby/Ruby23-x64/lib/ruby/site_ruby/2.3.0/rubygems/util/list.rb
   19 C:/Dev/ruby/Ruby23-x64/lib/ruby/2.3.0/x64-mingw32/stringio.so
   20 C:/Dev/ruby/Ruby23-x64/lib/ruby/site_ruby/2.3.0/rubygems/specification.rb
   21 C:/Dev/ruby/Ruby23-x64/lib/ruby/site_ruby/2.3.0/rubygems/exceptions.rb
   22 C:/Dev/ruby/Ruby23-x64/lib/ruby/site_ruby/2.3.0/rubygems/defaults/operating_system.rb
   23 C:/Dev/ruby/Ruby23-x64/lib/ruby/site_ruby/2.3.0/rubygems/dependency.rb
   24 C:/Dev/ruby/Ruby23-x64/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_gem.rb
   25 C:/Dev/ruby/Ruby23-x64/lib/ruby/2.3.0/monitor.rb
   26 C:/Dev/ruby/Ruby23-x64/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb
   27 C:/Dev/ruby/Ruby23-x64/lib/ruby/site_ruby/2.3.0/rubygems.rb
   28 C:/Dev/ruby/Ruby23-x64/lib/ruby/site_ruby/2.3.0/rubygems/path_support.rb
   29 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/did_you_mean-1.0.0/lib/did_you_mean/version.rb
   30 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/did_you_mean-1.0.0/lib/did_you_mean/core_ext/name_error.rb
   31 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/did_you_mean-1.0.0/lib/did_you_mean/levenshtein.rb
   32 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/did_you_mean-1.0.0/lib/did_you_mean/jaro_winkler.rb
   33 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/did_you_mean-1.0.0/lib/did_you_mean/spell_checkable.rb
   34 C:/Dev/ruby/Ruby23-x64/lib/ruby/2.3.0/delegate.rb
   35 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/did_you_mean-1.0.0/lib/did_you_mean/spell_checkers/name_error_checkers/class_name_checker.rb
   36 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/did_you_mean-1.0.0/lib/did_you_mean/spell_checkers/name_error_checkers/variable_name_checker.rb
   37 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/did_you_mean-1.0.0/lib/did_you_mean/spell_checkers/name_error_checkers.rb
   38 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/did_you_mean-1.0.0/lib/did_you_mean/spell_checkers/method_name_checker.rb
   39 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/did_you_mean-1.0.0/lib/did_you_mean/spell_checkers/null_checker.rb
   40 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/did_you_mean-1.0.0/lib/did_you_mean/formatter.rb
   41 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/did_you_mean-1.0.0/lib/did_you_mean.rb
   42 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d/exceptions.rb
   43 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d/color.rb
   44 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d/window.rb
   45 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d/application.rb
   46 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d/dsl.rb
   47 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d/quad.rb
   48 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d/rectangle.rb
   49 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d/square.rb
   50 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d/triangle.rb
   51 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d/image.rb
   52 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d/sprite.rb
   53 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d/text.rb
   54 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d/sound.rb
   55 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d/music.rb
   56 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d/ruby2d.so
   57 C:/Dev/ruby/Ruby23-x64/lib/ruby/gems/2.3.0/gems/ruby2d-0.3.0/lib/ruby2d.rb

[NOTE]
You may have encountered a bug in the Ruby interpreter or extension libraries.
Bug reports are welcome.
For details: http://www.ruby-lang.org/bugreport.html


This application has requested the Runtime to terminate it in an unusual way.
Please contact the application's support team for more information.
```

**Comment by KillaPL** [comment]
Make sure you have a font on that path, with that name. Seems like snippet from my comments, and I downloaded that font from the internet. There is ongoing pull request to allow writing texts without specifying font, and ongoing issue to guard against initialising texts and images without font.

**Comment by jjgh** [comment]
@KillaPL  It is snippet from your comments :) but I thought that running Text.new() with no arguments proved that the problem was not the missing font or wrong font path.
And yeah, I saw the pull request and that's a great idea.

Concerning guarding against initializing text and images without font, I thought it really was not another face of the same problem, because for example we can initialize a Square object without arguments, can't we? So I thought Text.new worked the same way.

Anyway, thanks for your answer!

**Comment by KillaPL** [comment]
Should be that way, that Text.new should just produce something, but it needs access to that font for that. Square doesn't need any file, so there isn't anything to crash : )

**Comment by blacktm** [comment]
As @KillaPL mentioned, this is expected at the moment but will be resolved in #53.
